module_test.go

```diff
@@ -241,6 +243,187 @@ func TestModuleSuccess(t *testing.T) {
+					"LoggerInitialized", "Invoking", "Invoked",
+				},
+			},
+			{
+				desc:           "Not using a custom logger for module defaults to app logger",
+				giveWithLogger: fx.Options(),
+				wantEventTypes: []string{
+					"Supplied", "Provided", "Provided", "Provided", "Provided",
+					"LoggerInitialized", "Invoking", "Invoked", "Invoking", "Invoked",
+				},
+			},
+		}
+
+		for _, tt := range tests {
+			tt := tt
+			t.Run(tt.desc, func(t *testing.T) {
+				t.Parallel()
+				var spy fxlog.Spy
+
+				redis := fx.Module("redis",
+					fx.Provide(func() *Logger {
+						return &Logger{Name: "redis"}
+					}),
+					fx.Invoke(func(r *Logger) {
+						assert.Equal(t, "redis", r.Name)
+					}),
+					tt.giveWithLogger,
+				)
+
+				app := fxtest.New(t,
+					redis,
+					fx.Supply(&spy),
+					fx.WithLogger(func(spy *fxlog.Spy) fxevent.Logger {
+						return spy
+					}),
+					fx.Invoke(func(r *Logger) {
+						assert.Equal(t, "redis", r.Name)
+					}),
+				)
+
+				// events from module with a custom logger not logged in app logger
+				assert.Equal(t, tt.wantEventTypes, spy.EventTypes())
+
+				spy.Reset()
+				app.RequireStart().RequireStop()
+
+				require.NoError(t, app.Err())
+
+				assert.Equal(t, []string{"Started", "Stopped"}, spy.EventTypes())
+			})
+		}
+	})
+
+	type NamedSpy struct {
+		fxlog.Spy
+		name string
+	}
+
+	t.Run("decorator on module logger does not affect app logger", func(t *testing.T) {
+		t.Parallel()
+
+		appSpy := NamedSpy{name: "app"}
+		moduleSpy := NamedSpy{name: "redis"}
+
+		redis := fx.Module("redis",
+			fx.Provide(func() *Logger {
+				return &Logger{Name: "redis"}
+			}),
+			fx.Supply(&appSpy),
+			fx.Replace(&moduleSpy),
+			fx.WithLogger(func(spy *NamedSpy) fxevent.Logger {
+				assert.Equal(t, "redis", spy.name)
+				return spy
+			}),
+			fx.Invoke(func(r *Logger) {
+				assert.Equal(t, "redis", r.Name)
+			}),
+		)
+
+		app := fxtest.New(t,
+			redis,
+			fx.WithLogger(func(spy *NamedSpy) fxevent.Logger {
+				assert.Equal(t, "app", spy.name)
+				return spy
+			}),
+			fx.Invoke(func(r *Logger) {
+				assert.Equal(t, "redis", r.Name)
+			}),
+		)
+
+		assert.Equal(t, []string{
+			"Provided", "Supplied", "Replaced",
+			"LoggerInitialized", "Invoking", "Invoked",
+		}, moduleSpy.EventTypes())
+
+		assert.Equal(t, []string{
+			"Provided", "Provided", "Provided",
+			"LoggerInitialized", "Invoking", "Invoked",
+		}, appSpy.EventTypes())
+
+		appSpy.Reset()
+		app.RequireStart().RequireStop()
+
+		require.NoError(t, app.Err())
+
+		assert.Equal(t, []string{"Started", "Stopped"}, appSpy.EventTypes())
+	})
+
+	t.Run("module uses parent module's logger to log events", func(t *testing.T) {
+		t.Parallel()
+
+		appSpy := NamedSpy{name: "app"}
+		childSpy := NamedSpy{name: "child"}
+
+		grandchild := fx.Module("grandchild",
+			fx.Provide(func() *Logger {
+				return &Logger{Name: "grandchild"}
+			}),
+			fx.Invoke(func(r *Logger) {
+				assert.Equal(t, "grandchild", r.Name)
+			}),
```

Comment:
I had to stare at this test for a while because it wasn't clear what this Logger thing was. Looks like this is just a random dependency that gets Provided to make sure that the "Provided" event gets logged by the loggers you supply, which are  `appSpy` and `childSupply`.

Can you change this so that it uses some other naive looking type, and not something that looks like an actual logger that gets passed to fx.WithLogger?